clients/src/main/java/org/apache/kafka/common/PartitionInfo.java

```diff
@@ -20,26 +20,35 @@
  * This is used to describe per-partition state in the MetadataResponse.
  */
 public class PartitionInfo {
+    public static final int UNKNOWN_LEADER_EPOCH = -1;
     private final String topic;
     private final int partition;
     private final Node leader;
+    private final int leaderEpoch;
     private final Node[] replicas;
     private final Node[] inSyncReplicas;
     private final Node[] offlineReplicas;
 
     public PartitionInfo(String topic, int partition, Node leader, Node[] replicas, Node[] inSyncReplicas) {
-        this(topic, partition, leader, replicas, inSyncReplicas, new Node[0]);
+        this(topic, partition, leader, UNKNOWN_LEADER_EPOCH, replicas, inSyncReplicas, new Node[0]);
+    }
+
+    public PartitionInfo(String topic, int partition, Node leader, Node[] replicas, Node[] inSyncReplicas,
+        Node[] offlineReplicas) {
+        this(topic, partition, leader, UNKNOWN_LEADER_EPOCH, replicas, inSyncReplicas, offlineReplicas);
     }
 
     public PartitionInfo(String topic,
                          int partition,
                          Node leader,
+                         int leaderEpoch,
```

Comment:
This is where we might get into trouble since we're breaking a constructor by adding a parameter.